### PR TITLE
Rebar Plugin Enhancement: Coverage copying of models

### PR DESCRIPTION
**Pull request by stevepm**
This enhancement checks if coverage is enabled in the rebar config. If it is AND the models directory is outside of the src directory, during the `eunit` hook, copy the models into the `.eunit` directory to allow coverage to run.

Problem this is solving:
We store our models in `models` directory outside of `src`. Eunit copies all of the erl files from src into the `.eunit` directory to allow coverage to run against them. But it doesn't take into account our models which are outside of `src`. This fixes that issue for us.

### CODE REVIEW

File: priv/rebar/boss_db_rebar.erl [hunk 1]
```diff
@@ -99,5 +113,50 @@ compile_model(Source, Target, BossDbOpts, RebarConfig) ->
...
+coverage_modifications_needed(RebarConf, BossDbOpts) ->
+    ModelDirIsNotSrc = model_dir_is_not_src(BossDbOpts),
+    coverage_enabled(RebarConf) and ModelDirIsNotSrc.
+
+%% @private
+%% @doc We only want to copy the models if they are stored outside of
+%% the src directory.
+model_dir_is_not_src(BossDbOpts) ->
+    ModelsDir = option(model_dir, BossDbOpts),
+    [FirstDir | _] = string:tokens(ModelsDir, "/"),
```
Comment (wk8) [inline priv/rebar/boss_db_rebar.erl]: `filename:split/1`

File: priv/rebar/boss_db_rebar.erl [hunk 2]
```diff
@@ -99,5 +113,50 @@ compile_model(Source, Target, BossDbOpts, RebarConfig) ->
...
+%% erlang file
+is_erl_file(Filename) ->
+    filename:extension(Filename) =:= ".erl".
+
+%% @private
+%% @doc Modifications are only needed if both coverage is enabled
+%% and the models are not in the src directory.
+coverage_modifications_needed(RebarConf, BossDbOpts) ->
+    ModelDirIsNotSrc = model_dir_is_not_src(BossDbOpts),
+    coverage_enabled(RebarConf) and ModelDirIsNotSrc.
```
Comment (wk8) [inline priv/rebar/boss_db_rebar.erl]: Kinda weird to use a straight fun call for one and assign to a var fro the other? :)

File: priv/rebar/boss_db_rebar.erl [hunk 3]
```diff
@@ -99,5 +113,50 @@ compile_model(Source, Target, BossDbOpts, RebarConfig) ->
...
+    ).
+
+%% @private
+copy_model_to_eunit_dir(Filename, ModelDir) ->
+    %% We only want to copy erl files. If it's something else,
+    %% don't touch it.
+    case is_erl_file(Filename) of
+        true ->
+            Source = filename:join([ModelDir, Filename]),
+            Destination = filename:join([".eunit", Filename]),
```
Comment (wk8) [inline priv/rebar/boss_db_rebar.erl]: filename:join/2 is better if 2 arguments :)

**Comment by danikp** [comment]
please update your code with latest master branch and comments from @wk8. Also, please keep in mind that support for rebar2 is going away and it's better to move to rebar3 plugin there you can set dirs for models

**Comment by wk8** [comment]
@danikp if you had a few minutes to have a look at this again, would be deeply appreciated :)

**Comment by danikp** [comment]
I don't have a problem with your approach, but boss_db moving to rebar3 and all that rebar-related code meant to be removed. Instead, you should take a look at rebar3_boss_db plugin https://github.com/danikp/rebar3_boss_db

**Pull request merged**